src/transformations/execute.ts

```diff
@@ -161,38 +162,64 @@ export async function executeTransformations(
 
       case 'macro':
         const recordedMacro = (await Register.get(transformation.register))?.text;
-        if (!(recordedMacro instanceof RecordedState)) {
+        if (!recordedMacro) {
           return;
-        }
 
-        vimState.isReplayingMacro = true;
+        } else if (typeof recordedMacro === 'string') {
+          // A string was set to the register. We need to execute the characters as if they were typed (in normal mode).
+          const keystrokes = keystrokesExpressionParser.parse(recordedMacro);
+          if (!keystrokes.status) {
+            return;
```

Comment:
Should this be communicated to the user somehow?

Reply:
See updated